modules/core/src/test/java/org/apache/ignite/internal/ClusterNodeMetricsSelfTest.java

```diff
@@ -75,7 +96,84 @@
+        offHeapValuesCfg.setStatisticsEnabled(true);
+        offHeapValuesCfg.setMemoryMode(CacheMemoryMode.OFFHEAP_VALUES);
+        offHeapValuesCfg.setOffHeapMaxMemory(MAX_VALS_AMOUNT * VAL_SIZE);
+
+        CacheConfiguration<Integer, Object> onHeapTieredCfg = defaultCacheConfiguration();
+        onHeapTieredCfg.setName(ON_HEAP_TIERED_NAME);
+        onHeapTieredCfg.setStatisticsEnabled(true);
+        onHeapTieredCfg.setMemoryMode(CacheMemoryMode.ONHEAP_TIERED);
+        onHeapTieredCfg.setOffHeapMaxMemory(MAX_VALS_AMOUNT * VAL_SIZE);
+
+        FifoEvictionPolicy plc = new FifoEvictionPolicy();
+        plc.setMaxMemorySize(MAX_VALS_AMOUNT * VAL_SIZE);
+        plc.setMaxSize(0);
+
+        onHeapTieredCfg.setEvictionPolicy(plc);
+
+        return cfg.setCacheConfiguration(offHeapValuesCfg, onHeapTieredCfg);
+    }
+
+    /**
+     *
+     */
+    public void testAllocatedMemory() throws Exception {
+        Ignite ignite = grid();
+
+        final IgniteCache onHeapCache = ignite.getOrCreateCache(ON_HEAP_TIERED_NAME);
+        final IgniteCache offHeapCache = ignite.getOrCreateCache(OFF_HEAP_VALUE_NAME);
+
+        long prevTieredOffHeapSize = onHeapCache.metrics().getOffHeapAllocatedSize();
+        long prevValuesOffHeapSize = offHeapCache.metrics().getOffHeapAllocatedSize();
+
+        assertEquals(0, prevTieredOffHeapSize);
+        assertEquals(0, prevValuesOffHeapSize);
+
+        long prevClusterNonHeapMemoryUsed = ignite.cluster().metrics().getNonHeapMemoryUsed();
+
+        fillCache(onHeapCache, getTestTimeout());
+
+        assertTrue(onHeapCache.metrics().getOffHeapAllocatedSize() > (MAX_VALS_AMOUNT - 5) * VAL_SIZE + prevTieredOffHeapSize);
+        assertEquals(0, offHeapCache.metrics().getOffHeapAllocatedSize());
+
+        // prevClusterNonHeapMemoryUsed = 69_277_368
+        // (MAX_VALS_AMOUNT - 5) * VAL_SIZE = 49_807_360
+        // tiered.metrics().getOffHeapAllocatedSize() = 51_012_531
+
+        assertTrue(prevClusterNonHeapMemoryUsed < ignite.cluster().metrics().getNonHeapMemoryUsed());
+
+        prevClusterNonHeapMemoryUsed = ignite.cluster().metrics().getNonHeapMemoryUsed();
+        prevTieredOffHeapSize = onHeapCache.metrics().getOffHeapAllocatedSize();
+
+        fillCache(offHeapCache, getTestTimeout());
+
+        assertTrue(offHeapCache.metrics().getOffHeapAllocatedSize() > (MAX_VALS_AMOUNT - 5) * VAL_SIZE);
+        assertEquals(prevTieredOffHeapSize, onHeapCache.metrics().getOffHeapAllocatedSize());
+        assertTrue((MAX_VALS_AMOUNT - 5) * VAL_SIZE + prevClusterNonHeapMemoryUsed < ignite.cluster().metrics().getNonHeapMemoryUsed());
+    }
+
+    /** Fill cache with values. */
+    private static void fillCache(final IgniteCache<Integer, Object> cache, long timeout) throws Exception{
+        final int THREAD_COUNT = 4;
+        final byte[] val = new byte[VAL_SIZE];
+        final AtomicInteger keyStart = new AtomicInteger(0);
+
+        GridTestUtils.runMultiThreaded(new Callable<Void>() {
```

Comment:
I don't see an significant reason why we need to initialize caches this way. Usually in tests caches are initialized with "cache.put" or "cache.putAll" because the data set is small. Follow the same approach.